auto_dev/commands/run.py

```diff
@@ -95,24 +115,52 @@ def fetch_agent(self) -> None:
 
     def setup_agent(self) -> None:
         """Setup the agent."""
-
-        self.logger.info(f"Agent author: {self.agent_name.author}")
-        self.logger.info(f"Agent name: {self.agent_name.author}")
-
-        self.change_directory(self.agent_name.name)
-        self.manage_keys()
-        self.install_dependencies()
-        self.issue_certificates()
-        self.logger.info("Agent setup complete. 🎉")
+        
+        try:
+            if not self.no_fetch:
+                self.logger.info(f"Agent author: {self.agent_name.author}")
+                self.logger.info(f"Agent name: {self.agent_name.name}")
+            
+            self.logger.info("Setting up agent keys...")
+            self.manage_keys()
+            
+            self.logger.info("Installing dependencies...")
+            self.install_dependencies()
+            
+            self.logger.info("Setting up certificates...")
+            self.issue_certificates()
+            
+            self.logger.info("Agent setup complete. 🎉")
+        except Exception as e:
+            self.logger.error(f"Failed to setup agent: {e}")
+            sys.exit(1)
 
     def manage_keys(self) -> None:
         """Manage Ethereum keys."""
-        if not Path("../ethereum_private_key.txt").exists():
-            self.execute_command("aea generate-key ethereum")
-            self.execute_command("aea -s add-key ethereum")
-        else:
+        self.logger.info("Checking for existing ethereum key...")
+        
+        
+        try:
+            result = subprocess.run(["aea", "-s", "remove-key", "ethereum"], capture_output=True, text=True)
+            if "no key registered" not in result.stderr.lower():
+                self.logger.info("Removed existing ethereum key")
+        except Exception:
+            pass 
+        
+        
+        if Path("ethereum_private_key.txt").exists():
+            self.logger.info("Found ethereum key in current directory")
+            self.execute_command("aea -s add-key ethereum ethereum_private_key.txt")
+        elif Path("../ethereum_private_key.txt").exists():
+            self.logger.info("Found ethereum key in parent directory, copying...")
             self.execute_command("cp ../ethereum_private_key.txt ./ethereum_private_key.txt")
-            self.execute_command("aea -s add-key ethereum")
+            self.execute_command("aea -s add-key ethereum ethereum_private_key.txt")
+        else:
+            self.logger.info("No existing ethereum key found, generating new one...")
+            self.execute_command("aea -s generate-key ethereum")
+            self.execute_command("aea -s add-key ethereum ethereum_private_key.txt")
```

Comment:
id include this block as seperate method, and pass the value of the SupportedLedger type

Reply:
we should have a default of ethereum imo but allow others as a param, imagine run will not be used in prod

Reply:
will update to use the default ledger param

Reply:
from the aea config